aif360/algorithms/inprocessing/adversarial_debiasing.py

```diff
@@ -177,14 +179,15 @@ def fit(self, dataset):
 
             if self.debias:
                 # Update adversary parameters
-                adversary_minimizer = adversary_opt.minimize(pred_protected_attributes_loss, var_list=adversary_vars, global_step=global_step)
+                with tf.control_dependencies([classifier_minimizer]):
+                    adversary_minimizer = adversary_opt.minimize(pred_protected_attributes_loss, var_list=adversary_vars)#, global_step=global_step)
 
             self.sess.run(tf.global_variables_initializer())
             self.sess.run(tf.local_variables_initializer())
 
             # Begin training
             for epoch in range(self.num_epochs):
-                shuffled_ids = np.random.choice(num_train_samples, num_train_samples)
+                shuffled_ids = np.random.choice(num_train_samples, num_train_samples, replace=False)
```

Comment:
I guess the changes to this function are to aid reproducibility? The replace=False seems to ensure that the batches will be mutually exclusive.

Reply:
Yes, the changes above were to aid reproducibility. `replace=False` will choose without replacement so the batches will be mutually exclusive, yes. That's more common for mini-batches so I figured it was an oversight although it works fine either way. @pronics2004, can you chime in if there was a reason you did it with replacement originally?